drivers/scheduler/dcos/marathon_ops.go

```diff
@@ -110,9 +141,31 @@ func (m *marathonOps) WaitForApplicationTermination(name string) error {
 			return nil, err
 		}
 
-		if _, err := m.client.Application(name); err != nil {
-			if matched, _ := regexp.MatchString(".+ does not exist", err.Error()); !matched {
-				return nil, err
+		_, err := m.client.Application(name)
+		if err == nil {
+			return nil, fmt.Errorf("Application %v is not yet deleted", name)
+		}
+
+		if matched, _ := regexp.MatchString(".+ does not exist", err.Error()); !matched {
```

Comment:
why regex? You could string.Contains. AFAIK Go's string.Contains is faster than its regex matching.

Reply:
done